rocky/rocky/views/ooi_detail.py

```diff
@@ -26,18 +26,21 @@ class OOIDetailView(BaseOOIDetailView, OOIRelatedObjectManager, OOIFindingManage
     def post(self, request, *args, **kwargs):
         if self.action == self.CHANGE_CLEARANCE_LEVEL:
             self.set_clearance_level()
-        if self.action == self.SUBMIT_ANSWER:
+        elif self.action == self.SUBMIT_ANSWER:
             self.answer_ooi_questions()
-        if self.action == self.START_SCAN:
+        elif self.action == self.START_SCAN:
             self.start_boefje_scan()
         return super().post(request, *args, **kwargs)
 
     def set_clearance_level(self) -> None:
         if not self.indemnification_present:
             return self.indemnification_error()
         else:
-            clearance_level = int(self.request.POST.get("level"))
-            self.can_raise_clearance_level(self.ooi, clearance_level)  # returns appropriate messages
+            try:
+                clearance_level = int(self.request.POST.get("level", ""))
+                self.raise_clearance_level(self.ooi, clearance_level)  # returns appropriate messages
+            except (ValueError, TypeError):
+                messages.error(self.request, _("There was an error setting this clearance level, not a number."))
```

Comment:
```suggestion
            try:
                clearance_level = int(self.request.POST["level"]))
                self.raise_clearance_level(self.ooi, clearance_level)  # returns appropriate messages
            except (ValueError, KeyError):
                messages.error(self.request, _("Cannot set clearance level. It must be provided and must be a valid number."))
```

I think this is a better approach, since there are two possible issues: the parameter might be missing or it might not be a valid number. Using `request.POST.get` would mask the missing parameter, making it harder to distinguish between the two cases